src/NHapi.Base/Parser/XMLParser.cs

```diff
@@ -348,9 +357,20 @@ public virtual bool Encode(ISegment segmentObject, XmlElement segmentElement)
                 for (var j = 0; j < reps.Length; j++)
                 {
                     var newNode = segmentElement.OwnerDocument.CreateElement(name);
+
                     var componentHasValue = Encode(reps[j], newNode);
                     if (componentHasValue)
                     {
+                        try
```

Comment:
rather than the use of `try catch` you could do the following...

```csharp
if (parserOptions.IncludeLongNameInEncodedXML && reps[j] is NHapi.Base.Model.AbstractType rep)
{
    newNode.SetAttribute("LongName", rep.Description);
}
```